ChopChop/Storage/StorageManager.swift

```diff
@@ -216,16 +219,29 @@ extension StorageManager {
     static let ingredientFolderName = "Ingredient"
     static let recipeFolderName = "Recipe"
 
-    func deleteRecipeImage(name: String) {
+    func deleteRecipeImage(name: String, id: Int64) throws {
+        guard var recipe = try fetchRecipe(id: id) else {
+            return
+        }
+        recipe.isImageUploaded = nil
+        try saveRecipe(&recipe)
         ImageStore.delete(imageNamed: name, inFolderNamed: StorageManager.recipeFolderName)
     }
 
     func fetchRecipeImage(name: String) -> UIImage? {
         ImageStore.fetch(imageNamed: name, inFolderNamed: StorageManager.recipeFolderName)
     }
 
-    func saveRecipeImage(_ image: UIImage, name: String) throws {
+    func saveRecipeImage(_ image: UIImage, id: Int64, name: String) throws {
         do {
+            let originalImage = fetchRecipeImage(name: name)
+            let isSameImage = originalImage?.pngData() == image.pngData()
+            guard var recipe = try fetchRecipe(id: id), !isSameImage else {
+                return
+            }
+            recipe.isImageUploaded = false
+            try saveRecipe(&recipe)
```

Comment:
I can move it to RecipeFormViewModel, but i will still need to reload the original image because it seems that the new image overwrites the image in RecipeFormViewModel (correct me if im wrong)

Reply:
But if there is a new image, doesn't that just mean you should change the `isImageUploaded` variable to true in the new recipe that is saved?